File: app/models/IOrder.java
package models;

import java.util.Date;
import java.util.List;

/**
 * Created by Beemen on 29/07/2015.
 */
public interface IOrder {
    String OrderUniqueID();
    void OrderUniqueID(String value);

    String OrderTitle();
    void OrderTitle(String value);

    // Get
    public List<OrderArchiveReference> Archives();

    List<DeliveryMethod> OrderOrigins();

    List<String > EndUserNotes();

    List<String > ArchivistNotes();

    Date IssueDate();
    void IssueDate(Date value);

    IEndUser User();
    void User(IEndUser value);

    Date PlannedDate();
    void PlannedDate(Date value);

    DateText ExpectedReadyDate();
    void ExpectedReadyDate(DateText value);

    IArchivist Archivist();
    void Archivist(IArchivist value);

    List<String> InternalNotes();

    List<DeliveryFormat> DeliveryFormats();
}

File: Models/IOrder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace end_user_gui.Models
{
    public interface IOrder
    {
        String OrderUniqueID { get; set; }
        String OrderTitle { get; set; }

        // Get
        List<OrderArchiveReference> Archives { get; }
        List<DeliveryMethod> OrderOrigins { get; }
        List<String> EndUserNotes { get; }
        List<String> ArchivistNotes { get; }
        DateTime IssueDate { get; set; }
        EndUser User { get; set; }
        DateTime PlannedDate { get; set; }
        DateText ExpectedReadyDate { get; set; }
        Archivist Archivist { get; set; }
        List<String> InternalNotes { get; }
        List<DeliveryFormat> DeliveryFormats { get; }

        /*Accessor methods*/
        Boolean Add(Archive archive);
        Boolean Add(Archive archive, Dissemination dissemination);
        Boolean Remove(Archive archive);
    }
}